### PR TITLE
Implemented: Gitbook lens search feature

### PR DESCRIPTION
**Pull request by amansinghbais**
### Related Issues

Closes #

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/company#contribution-guideline)

### CODE REVIEW

File: src/services/UtilService.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import api from "@/api"
+import api, {client} from "@/api"
+import { hasError } from "@/utils";
```
Comment (ymaheshwari1) [inline src/services/UtilService.ts:2]: Remove unused imports.

File: src/services/UtilService.ts [hunk 2]
```diff
@@ -40,10 +41,67 @@ const fetchShipmentMethodTypes = async (payload: any): Promise <any> => {
   });
 }
 
+const askQuery = async (payload: any): Promise <any> => {
+  const token = process.env.VUE_APP_GITBOOK_API_KEY;
+  const baseURL = `https://api.gitbook.com/v1/spaces/${process.env.VUE_APP_SPACE_ID}/search/`
+
+  return await client({
+    url: `ask`,
```
Comment (ymaheshwari1) [inline src/services/UtilService.ts:49]:
```suggestion
    url: "ask" 
```

File: src/services/UtilService.ts [hunk 3]
```diff
@@ -40,10 +41,67 @@ const fetchShipmentMethodTypes = async (payload: any): Promise <any> => {
...
+    url: `ask`, 
+    method: "post",
+    baseURL,
+    data: {
+      "query": payload.queryString
+    },
+    headers: {
+      Authorization:  'Bearer ' + token,
+      "Content-Type": "application/json",
+      "Access-Control-Allow-Origin":   "*"
```
Comment (ymaheshwari1) [inline src/services/UtilService.ts:58]: I think this is not required.

File: src/services/UtilService.ts [hunk 4]
```diff
@@ -40,10 +41,67 @@ const fetchShipmentMethodTypes = async (payload: any): Promise <any> => {
...
+    }
+  }) as any;
+}
+
+const getGitboookPage = async (pageId: any): Promise <any> => {
```
Comment (ymaheshwari1) [inline src/services/UtilService.ts:63]:
```suggestion
const getGitBookPage = async (pageId: any): Promise <any> => {
```

File: src/services/UtilService.ts [hunk 5]
```diff
@@ -40,10 +41,67 @@ const fetchShipmentMethodTypes = async (payload: any): Promise <any> => {
...
+  const token = process.env.VUE_APP_GITBOOK_API_KEY;
+  const baseURL = `https://api.gitbook.com/v1/spaces/${process.env.VUE_APP_SPACE_ID}/`
```
Comment (ymaheshwari1) [inline src/services/UtilService.ts:65]: We can keep the gitbook api endpoint in the env `https://api.gitbook.com/v1/spaces/`

```suggestion
  const baseURL = `https://api.gitbook.com/v1/spaces/${process.env.VUE_APP_SPACE_ID}/`
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/GitBookSearch.vue [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+  } catch(error: any) {
+    logger.error(error);
+  }
+
+  answer.value = response;
+  isQueryFetching.value = false;
+}
+
+async function searchQuery() {
+  isQueryFetching.value = true;
```
Comment (ymaheshwari1) [inline src/components/GitBookSearch.vue:191]:
```suggestion
  isLoading.value = true;
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Comment by amansinghbais** [comment]
Has implemented this with different approach in following PR : https://github.com/hotwax/company/pull/26

**Pull request closed without merging**